File: docs/DSA/neetcode150/src/go/two_sum.go
package main

func twoSum(nums []int, target int) []int {
	idxMap := make(map[int]int)
	for idx, num := range nums {
		if val, ok := idxMap[num]; ok {
			return []int{val, idx}
		}
		idxMap[target-num] = idx
	}
	return []int{}
}

File: DSA/neetcode150/src/py/two_sum.py
from typing import List
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        s = dict()
        for i, num in enumerate(nums):
            if num in s:
                return [i, s[num]]
            s[target - num] = i
        return [-1, -1]